baseplate/events/queue.py

```diff
@@ -0,0 +1,139 @@
+"""Client library for sending events to the data processing system.
+
+This is for use with the event collector system. Events generally track
+something that happens in production that we want to instrument for planning
+and analytical purposes.
+
+Events are serialized and put onto a message queue on the same server. These
+serialized events are then consumed and published to the remote event collector
+by a separate daemon.
+
+See also: https://github.com/reddit/event-collector
+
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import calendar
+import json
+import time
+import uuid
+
+from . import MAX_EVENT_SIZE, MAX_QUEUE_SIZE
+from baseplate.message_queue import MessageQueue, TimedOutError
+
+
+class Event(object):
+    """An event."""
+    def __init__(self, topic, event_type, timestamp=None, id=None):
+        self.topic = topic
+        self.event_type = event_type
+        if timestamp:
+            if timestamp.tzinfo and timestamp.utcoffset().total_seconds() != 0:
+                raise ValueError("Timestamps must be in UTC")
+            self.timestamp = calendar.timegm(timestamp.timetuple()) * 1000
+        else:
+            self.timestamp = time.time() * 1000
+        self.id = id or uuid.uuid4()
+        self.payload = {}
+
+    def get_field(self, key, obfuscated=False):
+        """Get the value of a field in the event.
+
+        If the field is not present, :py:data:`None` is returned.
+
+        :param str key: The name of the field.
+        :param bool obfuscated: Whether to look for the field in the obfuscated
+            payload.
+
+        """
+
+        if not obfuscated:
+            payload = self.payload
+        else:
+            payload = self.payload.get("obfuscated_data", {})
+        return payload.get(key, None)
+
+    def set_field(self, key, value, obfuscate=False):
+        """Set the value for a field in the event.
+
+        :param str key: The name of the field.
+        :param value: The value to set the field to. Should be JSON
+            serializable.
+        :param bool obfuscate: Whether or not to put the field in the obfuscated
+            section. This is used for sensitive info like IP addresses that must
+            be treated with care.
+
+        """
+        if not obfuscate:
+            self.payload[key] = value
```

Comment:
Currently the [eventcollector](https://github.com/reddit/reddit-public/blob/master/r2/r2/lib/eventcollector.py#L1146) checks to see if the `value` is `None` and won't add it to the payload if it is. Changing this would send some empty fields that I don't think are meant to be in there (and lead to a lot more lines in the code to fix this).